compiler/src/org.graalvm.compiler.lir/src/org/graalvm/compiler/lir/gen/LIRGeneratorTool.java

```diff
@@ -155,6 +155,15 @@ default boolean mayEmbedConstantLoad(Constant constant) {
 
     Value emitValueCompareAndSwap(LIRKind accessKind, Value address, Value expectedValue, Value newValue);
 
+    default Variable emitJavaLogicCompareAndSwap(LIRKind accessKind, Value address, Value expectedValue,
```

Comment:
I think it would be cleaner to pass a new flag into the existing methods instead of adding a whole new method.  Other platforms might ignore it but that's fine.

Reply:
Yes, in a sense it should be better. I was expecting to not have too much modifications for other platforms, especially the modifications to the common existing API. That's why I decided to add a new function here.

Reply:
Update the patch to use a new flag!